presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -141,38 +200,198 @@ public BaseJdbcClient(JdbcConnectorId connectorId, BaseJdbcConfig config, String
     }
 
     @Override
+    public Set<String> getSchemaNames()
+    {
+        ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+        Set<String> originalNames = getOriginalSchemas();
+        Map<String, Optional<String>> mappedNames = new HashMap<>();
+        for (String schemaName : originalNames) {
+            String schemaNameLower = schemaName.toLowerCase(ENGLISH);
+            schemaNames.add(schemaNameLower);
+            mappedNames.put(schemaNameLower, Optional.of(schemaName));
+        }
+
+        // if someone is listing all of the schema names, throw them all into the cache as a refresh since we already spent the time pulling them from the DB
+        schemaMappingCache.putAll(mappedNames);
```

Comment:
This method appears to be doing too much.  (1) it's computing all the schema names and (2) caching the values.

It's unclear where (2) is used and why (1) doesn't use it.

Furthermore, it forces that `reloadCache` must call `getSchemaNames` _before_ invalidating the values in `schemaTableMapping`.  This won't be clear to a future maintainer of the code.

Reply:
I don't use the cache because this method, in my view, must return the _current_ list of schema names directly from the DB. I also see this change as something which needs to NOT change the output of any method, and will only change the input, and even then as little as possible in order to get a query to execute.

The caching being done on the schema names is so that other requests for data can successfully execute those queries, even if it's just for listing the table names if the schema name has mixed case name.